pkg/models/networks.go

```diff
@@ -78,37 +78,72 @@ type CreateNetwork struct {
 	TypeID              int                  `json:"-" tf:"type_id"`
 	PoolID              int                  `json:"pool,omitempty" tf:"pool_id"`
 	NetworkDomainID     int                  `json:"-" tf:"domain_id"`
-	Site                IDStringModel        `json:"site,omitempty"`
 	Type                IDModel              `json:"type,omitempty"`
 	NetworkDomain       *IDModel             `json:"networkDomain,omitempty"`
 	NetworkProxy        *IDModel             `json:"networkProxy,omitempty"`
 	NetworkServer       IDModel              `json:"networkServer,omitempty"`
-	NetworkPool         PoolModel            `json:"networkPool,omitempty"`
+	NetworkPool         *PoolModel           `json:"networkPool,omitempty"`
 	NetworkProxyID      int                  `json:"-" tf:"proxy_id"`
 	ProxyID             int                  `json:"-" tf:"proxy_id"`
 	SearchDomains       string               `json:"searchDomains,omitempty" tf:"search_domains"`
 	Cidr                string               `json:"cidr,omitempty" tf:"cidr"`
 	Gateway             string               `json:"gateway,omitempty" tf:"gateway"`
 	DNSPrimary          string               `json:"dnsPrimary,omitempty" tf:"primary_dns"`
 	DNSSecondary        string               `json:"dnsSecondary,omitempty" tf:"secondary_dns"`
-	Config              *CreateNetworkConfig `json:"config,omitempty" tf:"config,sub"`
 	Active              bool                 `json:"active" tf:"active"`
-	DhcpServer          bool                 `json:"dhcpServer"`
 	ScanNetwork         bool                 `json:"scanNetwork" tf:"scan_network"`
 	AllowStaticOverride bool                 `json:"allowStaticOverride" tf:"allow_static_override"`
 	AppURLProxyBypass   bool                 `json:"applianceUrlProxyBypass,omitempty" tf:"appliance_url_proxy_bypass"`
 	NoProxy             string               `json:"noProxy,omitempty" tf:"no_proxy"`
 	ScopeID             string               `json:"scopeId,omitempty" tf:"scode_id"`
+	ExternalID          int                  `json:"externalId"`
+	InternalID          int                  `json:"internalId"`
+	UniqueID            int                  `json:"uniqueId"`
+	Status              string               `json:"status"`
+	Code                string               `json:"code"`
+	Group               IDStringModel        `json:"site" tf:"group"`
+	DhcpServer          bool                 `json:"dhcpServer"`
+	TfDhcpNetwork       *CreateDhcpNetwork   `json:"-" tf:"dhcp_network,sub"`
+	TfStaticNetwork     *CreateStaticNetwork `json:"-" tf:"static_network,sub"`
+	Config              CreateNetworkConfig  `json:"config"`
 	ResourcePermissions NetworkResPermission `json:"-" tf:"resource_permissions,sub"`
 }
 
+type CreateDhcpNetwork struct {
+	DhcpServer bool           `json:"-" tf:"dhcp_enabled"`
+	Config     *NetworkConfig `json:"-" tf:"config,sub"`
+}
+
+type CreateStaticNetwork struct {
+	ExternalID string         `json:"-" tf:"external_id,computed"`
```

Comment:
why do need these to be tf fields ?

Reply:
InternalID, Unique ID , External ID

Reply:
code as well.

Reply:
before  tf fields was in the line https://github.com/HewlettPackard/hpegl-vmaas-cmp-go-sdk/blob/77c6f87dd327ec7efa15bdaabf62e689d5e5f9a9/pkg/models/networks.go#L35 now I moved to static block